Client/Application/AppDelegate.swift

```diff
@@ -837,39 +837,41 @@ class AppSyncDelegate: SyncDelegate {
     }
 
     open func displaySentTab(for url: URL, title: String, from deviceName: String?) {
-        if let appDelegate = app.delegate as? AppDelegate, app.applicationState == .active {
-            DispatchQueue.main.async {
-                appDelegate.browserViewController.switchToTabForURLOrOpen(url, isPrivileged: false)
+        DispatchQueue.main.sync {
+            if let appDelegate = app.delegate as? AppDelegate, app.applicationState == .active {
+                DispatchQueue.main.async {
```

Comment:
Can this block go away since we already do that two lines up? At this point we are already executing on the main thread right?